controllers/controller.go

```diff
@@ -187,14 +190,13 @@ func (r *EtcdadmClusterReconciler) reconcile(ctx context.Context, etcdCluster *e
 			outdatedMachines := etcdMachines.Difference(ownedMachines)
 			log.Info(fmt.Sprintf("Controlplane upgrade has completed, deleting older outdated etcd members: %v", outdatedMachines.Names()))
 			for _, outdatedMachine := range outdatedMachines {
-				err := r.removeEtcdMemberAndDeleteMachine(ctx, etcdCluster, cluster, ep, outdatedMachine)
-				if len(outdatedMachines) > 1 {
+				err := r.removeEtcdMember(ctx, etcdCluster, cluster, ep, outdatedMachine)
```

Comment:
this is to ensure that the controller reconciles and deletes any outdated machines post-upgrade, so it either returns an error here if deletion of any machine fails, or returns nil while requeueing the object